--- LightService.h	2016-12-02 22:04:50.872906837 -0600
+++ /home/hunter/Arduino/hoopla/LightService.h	2016-12-02 23:53:52.004803756 -0600
@@ -1,3 +1,5 @@
+#include <ESP8266WebServer.h>
+
 enum HueColorType {
   TYPE_HUE_SAT, TYPE_CT, TYPE_XY
 };
@@ -42,7 +44,7 @@
       bool setLightsAvailable(int numLights);
       int getLightsAvailable();
       bool setLightHandler(int index, LightHandler *handler);
-      void begin();
+      void begin(ESP8266WebServer * svr);
       void update();
     private:
       int currentNumLights = MAX_LIGHT_HANDLERS;
